include/wally_psbt.h

```diff
@@ -122,338 +181,399 @@ struct wally_psbt {
- * :param witness_script: The witness script for this input
- * :param witness_script_len: The length of the witness script.
- * :param final_script_sig: The scriptSig for this input
- * :param final_script_sig_len: Size of ``final_script_sig`` in bytes.
+ * :param redeem_script: The redeem script for this input.
+ * :param redeem_script_len: Length of ``redeem_script`` in bytes.
+ * :param witness_script: The witness script for this input.
+ * :param witness_script_len: Length of ``witness_script`` in bytes.
+ * :param final_script_sig: The scriptSig for this input.
+ * :param final_script_sig_len: Length of ``final_script_sig`` in bytes.
  * :param final_witness: The witness stack for the input, or NULL if no witness is present.
  * :param keypaths: The HD keypaths for this input.
  * :param partial_sigs: The partial signatures for this input.
  * :param unknowns: The unknown key value pairs for this input.
- * :param sighash_type: The sighash type for this input
- * :param output: Destination for the resulting psbt input.
+ * :param sighash_type: The sighash type for this input.
+ * :param output: Destination for the resulting PSBT input.
  */
 WALLY_CORE_API int wally_psbt_input_init_alloc(
-    struct wally_tx *non_witness_utxo,
-    struct wally_tx_output *witness_utxo,
-    unsigned char *redeem_script,
+    const struct wally_tx *non_witness_utxo,
+    const struct wally_tx_output *witness_utxo,
+    const unsigned char *redeem_script,
     size_t redeem_script_len,
-    unsigned char *witness_script,
+    const unsigned char *witness_script,
     size_t witness_script_len,
-    unsigned char *final_script_sig,
+    const unsigned char *final_script_sig,
     size_t final_script_sig_len,
-    struct wally_tx_witness_stack *final_witness,
-    struct wally_keypath_map *keypaths,
-    struct wally_partial_sigs_map *partial_sigs,
-    struct wally_unknowns_map *unknowns,
+    const struct wally_tx_witness_stack *final_witness,
+    const struct wally_keypath_map *keypaths,
+    const struct wally_partial_sigs_map *partial_sigs,
+    const struct wally_unknowns_map *unknowns,
     uint32_t sighash_type,
     struct wally_psbt_input **output);
 
 /**
- * Set the non_witness_utxo in an input
+ * Set the non_witness_utxo in an input.
  *
  * :param input: The input to update.
  * :param non_witness_utxo: The non witness utxo for this input if it exists.
  */
 WALLY_CORE_API int wally_psbt_input_set_non_witness_utxo(
     struct wally_psbt_input *input,
-    struct wally_tx *non_witness_utxo);
+    const struct wally_tx *non_witness_utxo);
 
 /**
- * Set the witness_utxo in an input
+ * Set the witness_utxo in an input.
  *
  * :param input: The input to update.
  * :param witness_utxo: The witness utxo for this input if it exists.
  */
 WALLY_CORE_API int wally_psbt_input_set_witness_utxo(
     struct wally_psbt_input *input,
-    struct wally_tx_output *witness_utxo);
+    const struct wally_tx_output *witness_utxo);
```

Comment:
Should we mention in `CHANGES.md` that some API parameters are now const?

Reply:
Moving from non-const to const isn't an API change for the caller, although the other way around would be. So there's no need to mention this here I think.